src/Riok.Mapperly/Descriptors/MappingBuilders/ConvertStaticMethodMappingBuilder.cs

```diff
@@ -0,0 +1,204 @@
+        return ctx.Source.SpecialType == SpecialType.System_DateTime
+                && ctx.Types.TimeOnly != null
+                && ctx.Target is INamedTypeSymbol namedSymbol
+                && SymbolEqualityComparer.Default.Equals(namedSymbol, ctx.Types.TimeOnly)
+            || true;
+    }
+
+    private static bool TryGetStaticMethodMapping(
+        ICollection<IMethodSymbol> allMethods,
+        IEnumerable<string> methodNames,
+        ITypeSymbol sourceType,
+        ITypeSymbol targetType,
+        ITypeSymbol nonNullableTargetType,
+        bool targetIsNullable,
+        out StaticMethodMapping? mapping
+    )
+    {
+        foreach (var methodName in methodNames)
+        {
+            var candidates = GetMethodCandidates(allMethods, methodName, sourceType);
+
+            // try to find method with equal nullability return type
+            var method = candidates.FirstOrDefault(x => SymbolEqualityComparer.IncludeNullability.Equals(x.ReturnType, targetType));
+
+            if (method != null)
+            {
+                mapping = new StaticMethodMapping(method);
+                return true;
+            }
+
+            if (!targetIsNullable)
+                continue;
+
+            // otherwise try to find method ignoring the nullability
+            method = candidates.FirstOrDefault(x => SymbolEqualityComparer.Default.Equals(x.ReturnType, nonNullableTargetType));
+
+            if (method == null)
+            {
+                continue;
+            }
+
+            mapping = new StaticMethodMapping(method);
+            return true;
+        }
+
+        mapping = null;
+        return false;
+    }
+
+    private static IMethodSymbol[] GetMethodCandidates(ICollection<IMethodSymbol> allMethods, string methodName, ITypeSymbol parameterType)
+    {
+        return allMethods
+            .Where(m =>
+                string.Equals(m.Name, methodName, StringComparison.Ordinal)
+                && m is { IsStatic: true, ReturnsVoid: false, IsAsync: false, Parameters.Length: 1 }
+                && FilterParameterType(m.Parameters[0], parameterType)
+            )
+            .ToArray();
+    }
+
+    private static bool FilterParameterType(IParameterSymbol parameter, ITypeSymbol targetType)
+    {
+        if (SymbolEqualityComparer.Default.Equals(parameter.Type, targetType))
+            return true;
+
+        if (!parameter.IsParams)
+            return false;
+
+        var targetIsArray = targetType.IsArrayType(out var targetArrayTypeSymbol);
+
+        var targetIsEnumerable = targetType.AllInterfaces.Any(x =>
+            x.OriginalDefinition.SpecialType == SpecialType.System_Collections_Generic_IEnumerable_T
+        );
+
+        if (parameter.Type.IsArrayType(out var arrayTypeSymbol))
+        {
+            return SymbolEqualityComparer.Default.Equals(
+                arrayTypeSymbol.ElementType,
+                targetIsArray ? targetArrayTypeSymbol!.ElementType : targetType
+            );
+        }
```

Comment:
This early return could happen before the enumerable check of the target, couldn't it?